### PR TITLE
Unify p2p blocks processing (pubsub/sync)

### PR DESCRIPTION
**Pull request by stanislav-tkach**
Previously we had two separate mechanisms:
- Initial syncing was done by the `sync` module.
- After that new blocks were handled by the `pubsub` module.

The pull request unifies this logic by removing the `pubsub` module.

### CODE REVIEW

File: p2p/src/net/libp2p/service/syncing.rs [hunk 1]
```diff
@@ -139,6 +137,10 @@ where
                 request_id,
                 error,
             }),
+            types::SyncingEvent::Announcement { .. } => {
+                todo!();
+                todo!()
```
Comment (TheQuantumPhysicist) [inline p2p/src/net/libp2p/service/syncing.rs:142]: Wow... there's twice as much to do here :smile:

File: p2p/src/net/mock/backend.rs [hunk 2]
```diff
@@ -319,6 +333,10 @@ where
             .map_err(P2pError::from)
     }
 
+    async fn handle_announcement(&mut self, _announcement: Announcement) -> crate::Result<()> {
+        todo!();
```
Comment (TheQuantumPhysicist) [inline p2p/src/net/mock/backend.rs:337]: ?

File: p2p/src/lib.rs [hunk 3]
```diff
@@ -203,32 +203,13 @@ where
                     consensus_handle,
```
Comment (TheQuantumPhysicist) [inline p2p/src/lib.rs:203]: We're still calling this consensus for some reason... please rename it to chainstate_handle

File: p2p/src/net/mock/backend.rs (hunk 2 above)
Comment (stanislav-tkach) [reply p2p/src/net/mock/backend.rs:337]: The block announcement (pubsub) wasn't implemented for the mock backed, so in this pull request I have added the minimal required parts in order to pass the existing tests. I'm going to create an issue to finish this implementation.

Comment (stanislav-tkach) [reply p2p/src/net/mock/backend.rs:337]: https://github.com/mintlayer/mintlayer-core/issues/488

**Comment by TheQuantumPhysicist** [comment]
If I understand correctly, the test that wasn't working now is working. So now this is free to get merged.

**Review by TheQuantumPhysicist — APPROVED** [review]

**Pull request merged**